### ISSUE TITLE
[Bug][ComboBox] Cannot read properties of undefined during keyboard navigation

### ISSUE THREAD

**Issue by WissamProgress** [telerik/kendo-react#1765]
## I'm submitting a...

* Bug report

## Current behavior
When binding the ComboBox to an [array of objects](https://www.telerik.com/kendo-react-ui/components/dropdowns/combobox/binding/#toc-datasets-of-objects), and having only one item, pressing the ArrowDown key fires the following TypeError - `Cannot read properties of null (reading 'text')`
Pressing the ArrowUp button navigates to the item correctly.

## Expected behavior
The component does not fire any errors during keyboard navigation, and navigates correctly to the items regardless if there is only one item or more.

## Minimal reproduction of the problem with instructions

1. Open this example https://codesandbox.io/s/dank-microservice-cmvvgy
2. Press on the component
3. Press to ArrowDown key

## Workaround
As a workaround, set the data to an [array of primitive values](https://www.telerik.com/kendo-react-ui/components/dropdowns/combobox/binding/#toc-arrays-of-primitive-values) and do not set the `textField` and `dataItemKey` properties when the data length is less than 1:

https://codesandbox.io/s/pensive-blackburn-nnffjd?file=/app/main.tsx

Reported in Ticket ID: 1631463

**Comment by silviyaboteva** [comment]
This pull request is included in version 7.2.0.
